### ISSUE TITLE
Another issue with No files shown in folders mounted from NAS

### ISSUE THREAD

**Issue by 88Dit** [jlesage/docker-crashplan-pro#126]
Hi All,

I'm a first timer working with the qnap docker crashplan scenario.  I'm not very apt at its CLI, I'm more of a gui guy but I'm learning.

I'm running the app as follows:
docker run -d \
    --name=crashplan-pro \
    -e USER_ID=0 -e GROUP_ID=0 \
    -p 5800:5800 \
    -v /DataVol1/Container/Crashplan-Pro:/config:rw \
    -v /DataVol1/Staff:/storage:ro \
    jlesage/crashplan-pro

It is my understanding that I am mapping my /DataVol1/Staff folder to be accessible to the container as  Storage.  It is this Staff folder/subfolders that I need to back up.  (There are other folders, but let's tackle this issue first).

The container starts and is accessible, however there are no files listed with:
docker exec crashplan-pro ls -l /storage

8 files are listed, as expected, with:
docker exec crashplan-pro ls -l /config

From what I've read here and on the qnap forums, it might be a permissions issue.  However, I'm using the built-in admin account:
id=0(admin) gid=0(administrators) groups=0(administrators),100(everyone)

Which does have the appropriate permissions as shown in this screen shot:
![image](https://user-images.githubusercontent.com/42877656/44933822-75be9c00-ad38-11e8-9d80-c1227b2a49b1.png)

Trying to enumerate permission via CLI doesn't seem to work either:
[~] # ls -l /DataVol1/Staff/* && ls -l /DataVol1/Staff/*/*
/bin/ls: cannot access /DataVol1/Staff/*: No such file or directory

While I am replacing my Qnap TS-469 Pro with a TS-673 and moving off of CP Home to my existing CP Business account, I don't need to keep my backup so I am not adopting the existing device but rather creating a new device.  I don't care that it has to back up all files again as there isn't too much data and I don't need the history.

Everything has gone quite well and I'm pleased with the simplicity of it, however perhaps I'm overlooking something or don't have the proper parameters in place when running CrashPlan?  Please let me know what additional information I can provide.

**Comment by jlesage** [comment]
From what I understand, QNAP shares can be accessed from multiple paths.

I think you should also be able to access your share via `/share/CACHEDEV1_DATA/Staff`.  Can you try to map this path instead?

**Comment by 88Dit** [comment]
Thank you for your assistance.  Using the CACHEDEV1_DATA paths seems to be working.  Here is my current running command:
docker run -d \
    --name=crashplan-pro \
    -e USER_ID=0 -e GROUP_ID=0 \
    -p 5800:5800 \
    -v /share/CACHEDEV1_DATA/Containers/crashplan/config:/config:rw \
    -v /share/CACHEDEV1_DATA/:/storage:ro \
    jlesage/crashplan-pro

There were a few hangups along the way, both on the Crashplan side (waiting for backup) and also with the Qnap file system becoming inaccessible (which resulted in multiple containers/volumes and prompted a Qnap reboot and Raid rebuild), but all seems to be working fine for the moment.  I will continue my observations while Crashplan is going through its initial seeding.

**Comment by ning42** [comment]
If it is like me, it is because in the UI you can choose the /share/CACHEDEV1_DATA/ and mount that as /storage
 - you have to choose a shared folder :(
If you mount /share/CACHEDEV1_DATA/ then Crashplan can backup all folders/shares etc.

**Comment by gatorheel** [comment]
I can tell you why I finally decided to figure out the right command line (as well as persistent, ex-container config storage).  For some reason whenever Container Station (the QNAP docker app) upgrades, my Crashplan container breaks, and I have to destroy and rebuild it.  If I use the CLI, I can put the command sequence and arguments in a script and a) make it easy to replay and b) guarantee that I haven't forgotten a switch or config if it's been a while since the last rebuild.

**Comment by 88Dit** [comment]
Right, in this case, it is much easier for me to copy/paste a command line instead of running through a GUI where I might forget an option.

**Comment by jlesage** [comment]
Thanks for the explanation.  I guess I should add to the README the restrictions people need to follow when mapping folders (i.e. use `/share/CACHEDEV1_DATA/` or something under it).

**Comment by cerealsoup** [comment]
Having a similar issue. Wondering if there is an operating difference between these two mappings? Is one correct and are either valid?


`-v /share/Backup:/storage:ro` or `-v /share/Backup/:/storage:ro`

does the directory need the last slash to be in the docker run command?

**Comment by 88Dit** [comment]
> Having a similar issue. Wondering if there is an operating difference between these two mappings? Is one correct and are either valid?
> 
> `-v /share/Backup:/storage:ro` **OR** `-v /share/Backup/:/storage:ro`
> 
> does the directory need the last slash to be in the docker run command?

It runs both ways for me.  Are you using a QNAP? If so, try using:
 -v /share/CACHEDEV1_DATA :/storage:ro \ 

Then you should be able to browse to your shares via {device}\root\storage\

**Comment by 88Dit** [comment]
I've been testing this configuration for the past month, and occasionally see hiccups.  For example, when I stop the container and attempt to change its settings (auto-start, memory, cpu etc) I get the following error message:
![image](https://user-images.githubusercontent.com/42877656/46611140-e5b00700-cada-11e8-958c-f37d8fe1d43a.png)
Then there are two images, two volumes, and two containers
![image](https://user-images.githubusercontent.com/42877656/46611270-548d6000-cadb-11e8-954f-3e499f20f044.png)

![image](https://user-images.githubusercontent.com/42877656/46611234-3aec1880-cadb-11e8-8702-1c75928fa494.png)

The new container does not start. The 'temporary' container is the original one and starts and appears to be fine.  However I don't like having the temporary label in production, so I end up removing the container and recreating it.  Any reason why I can't change any settings to it?

**Comment by jlesage** [comment]
Sorry I can't really help you on this (I don't have a QNAP), but it seems to be a question that could be asked to QNAP support team and/or forum.
